genesyscloud/idp_generic/genesyscloud_idp_generic_proxy.go

```diff
@@ -0,0 +1,89 @@
+type deleteIdpGenericFunc func(ctx context.Context, p *idpGenericProxy, id string) (response *platformclientv2.APIResponse, err error)
+
+// idpGenericProxy contains all of the methods that call genesys cloud APIs.
+type idpGenericProxy struct {
+	clientConfig         *platformclientv2.Configuration
+	identityProviderApi  *platformclientv2.IdentityProviderApi
+	getIdpGenericAttr    getIdpGenericFunc
+	updateIdpGenericAttr updateIdpGenericFunc
+	deleteIdpGenericAttr deleteIdpGenericFunc
+}
+
+// newIdpGenericProxy initializes the idp generic proxy with all of the data needed to communicate with Genesys Cloud
+func newIdpGenericProxy(clientConfig *platformclientv2.Configuration) *idpGenericProxy {
+	api := platformclientv2.NewIdentityProviderApiWithConfig(clientConfig)
+	return &idpGenericProxy{
+		clientConfig:         clientConfig,
+		identityProviderApi:  api,
+		getIdpGenericAttr:    getIdpGenericFn,
+		updateIdpGenericAttr: updateIdpGenericFn,
+		deleteIdpGenericAttr: deleteIdpGenericFn,
+	}
+}
+
+// getIdpGenericProxy acts as a singleton to for the internalProxy.  It also ensures
+// that we can still proxy our tests by directly setting internalProxy package variable
+func getIdpGenericProxy(clientConfig *platformclientv2.Configuration) *idpGenericProxy {
+	if internalProxy == nil {
+		internalProxy = newIdpGenericProxy(clientConfig)
+	}
+
+	return internalProxy
+}
+
+// getIdpGeneric retrieves all Genesys Cloud idp generic
+func (p *idpGenericProxy) getIdpGeneric(ctx context.Context) (*platformclientv2.Genericsaml, *platformclientv2.APIResponse, error) {
+	return p.getIdpGenericAttr(ctx, p)
+}
+
+// updateIdpGeneric updates a Genesys Cloud idp generic
+func (p *idpGenericProxy) updateIdpGeneric(ctx context.Context, id string, idpGeneric *platformclientv2.Genericsaml) (*platformclientv2.Identityprovider, *platformclientv2.APIResponse, error) {
+	return p.updateIdpGenericAttr(ctx, p, id, idpGeneric)
+}
+
+// deleteIdpGeneric deletes a Genesys Cloud idp generic by Id
+func (p *idpGenericProxy) deleteIdpGeneric(ctx context.Context, id string) (resp *platformclientv2.APIResponse, err error) {
+	return p.deleteIdpGenericAttr(ctx, p, id)
+}
+
+// getIdpGenericFn is the implementation for retrieving all idp generic in Genesys Cloud
+func getIdpGenericFn(ctx context.Context, p *idpGenericProxy) (*platformclientv2.Genericsaml, *platformclientv2.APIResponse, error) {
+	return p.identityProviderApi.GetIdentityprovidersGeneric()
+}
+
+// updateIdpGenericFn is an implementation of the function to update a Genesys Cloud idp generic
+func updateIdpGenericFn(ctx context.Context, p *idpGenericProxy, id string, idpGeneric *platformclientv2.Genericsaml) (*platformclientv2.Identityprovider, *platformclientv2.APIResponse, error) {
+	generic, resp, err := p.identityProviderApi.PutIdentityprovidersGeneric(*idpGeneric)
+
+	return generic, resp, err
+}
```

Comment:
Similar thing here. You can just `return p.identityProviderApi.PutIdentityprovidersGeneric(*idpGeneric)` directly